### PR TITLE
Rename start & end properties to fix conflict with RN 51

### PR DESCRIPTION
**Pull request by bilalsyed001**
This works for both iOS and Android.

credit @sjmueller , @MichaelDanielTom

### CODE REVIEW

**Comment by terribleben** [comment]
It may have been this: https://github.com/facebook/react-native/commit/38b5506599b12d5c6ac3a2dd3f047ccf067cfc71

**Comment by sjmueller** [comment]
@jdonisvitch I explained in the original PR #235

**Comment by KamranKhankhail** [comment]
It must be merged, as soon as possible.

**Comment by Kottidev** [comment]
@dabit3

File: index.android.js [hunk 1]
```diff
@@ -55,9 +55,9 @@ export default class LinearGradient extends Component {
     const {
       children,
       colors,
-      end,
```
[the remark this answers is not in the document]
Comment (jdonisvitch) [reply index.android.js]: That is a really good point. It's not at the component level but at the internal level. Does the RN team believe this is a bug on their end?

[the remark this answers is not in the document]
Comment (koenpunt) [reply index.android.js]: Well I don't think having more explicit methods like `setStartPoint` and `setEndPoint` internally is an issue. So I'd say if only the internal implementation is updated, there's no fix need in React Native (if there ever is going to be one), and the library can just move forward.

**Comment by dannycochran** [comment]
Per comment in https://github.com/react-native-community/react-native-linear-gradient/pull/244, the index.d.ts typings file should also be updated to use "startPoint" and "endPoint"

**Comment by raphaeleidus** [comment]
This changes the API. this should not be merged. in #235  @sjmueller was able to address the issue without changing the external API. This is strongly preferred

**Comment by sjmueller** [comment]
I have updated my PR to support android, and it does not change the external API (as @raphaeleidus mentioned).

**Comment by kesha-antonov** [comment]
+++

**Comment by afilp** [comment]
Can you please merge one or the other? We cannot release due to this and we cannot go back from 0.51 because 0.50 had another bug that was causing problems and now it is fixed.

**Comment by hwangjr** [comment]
please merge this request, and release a new version, as soon as possible.  thanks. @koenpunt

**Comment by KarandikarMihir** [comment]
Any word on this? I just updated to 0.51 and I really don't want to go back :|

**Comment by krailler** [comment]
I need this fix... :/

**Comment by ryanhomer** [comment]
It looks like #235 was merged instead.

**Comment by mujavidb** [comment]
Any update on this?

**Comment by sibelius** [comment]
can we close this?

**Comment by abeddow91** [comment]
I'm getting this issue now with 

react-native-linear-gradient@2.5.2
react-native: 0.57.7

any ideas?

**Comment by Lcharvol** [comment]
up plz, 2021 and the problem persists

**Comment by alphonse92** [comment]
Still happening: "react-native-linear-gradient": "^2.5.6",

**Comment by alphonse92** [comment]
If you still have this issue you must create a HOC to rename the props as you can see in this example:

image

**Pull request closed without merging**